clang-tidy.sh

```diff
@@ -0,0 +1,25 @@
+#!/bin/sh
+
+set -e
+
+FIXES_FILE="tidy-fixes.yaml"
```

Comment:
Is this file even necessary now?

Reply:
It isn't needed for the CI test, I left it in for local tests, auto-fixing, ...

Reply:
But then give the user the ability to enable auto fixing. E.g. pass all given args to `run-clang-tidy`. So I can run `./clang-tidy.sh -fix` to fix my errors...